lib/Experian/IDAuth.pm

```diff
@@ -86,10 +89,8 @@ sub save_pdf_result {
         # Download pdf result on given reference number
         $mech->get("$url/archive/index.cfm?event=archive.pdf&id=$our_ref");
         1;
-    } || do {
-        my $err = $@;
-        warn "errors downloading pdf: $err";
-        return;
+    } catch {
+        croak "errors downloading pdf: $_";
```

Comment:
throwing exception again in catch? is it intented?

Reply:
Yes, explaining what failed with the error message.